scheds/rust/scx_layered/src/main.rs

```diff
@@ -2147,6 +2155,179 @@ impl<'a> Scheduler<'a> {
         weighted
     }
 
+    fn compute_target_llcs(target: usize, topo: &Topology) -> (usize, usize) {
+        // TODO(kkd): We assume each LLC has equal number of cores.
+        let cores_per_llc = topo.all_cores.len() / topo.all_llcs.len();
+        // TODO(kkd): We assume each core has fixed number of threads.
+        let cpus_per_core = topo.all_cores.first_key_value().unwrap().1.cpus.len();
+        let cpus_per_llc = cores_per_llc * cpus_per_core;
+
+        let full = target / cpus_per_llc;
+        let extra = target % cpus_per_llc;
+
+        (full, extra.div_ceil(cpus_per_core))
+    }
+
+    fn recompute_layer_core_order(&mut self, layer_targets: &Vec<(usize, usize)>) {
+        // Collect freed LLCs from shrinking layers.
+        debug!(
+            " free: before pass: free_llcs={:?}",
+            self.cpu_pool.free_llcs
+        );
+        for &(idx, target) in layer_targets.iter().rev() {
+            let layer = &mut self.layers[idx];
+            let old_tlc = layer.target_llc_cpus;
+            let new_tlc = Self::compute_target_llcs(target, &self.topo);
+
+            if let LayerGrowthAlgo::StickyDynamic = layer.growth_algo {
```

Comment:
Would adding `#[derive(PartialEq)]` to `enum LayerGrowthAlgo` and doing `if layer.growth_algo != LayerGrowthAlgo::StickyDynamic` work?

Reply:
Thanks for the suggestion, that works!